src/logic/wallets/utils/network.ts

```diff
@@ -20,7 +20,9 @@ const WALLET_ERRORS = {
 const requestSwitch = async (wallet: Wallet, chainId: ChainId): Promise<void> => {
   // Note: This could support WC too
   if (isPairingModule(wallet.name)) {
-    wallet.provider.wc.updateSession({ chainId: parseInt(chainId, 10), accounts: wallet.provider.wc.accounts })
+    if (wallet.provider) {
```

Comment:
This was throwing without the provider check. We could maybe remove it when we later persist Onboard cross-chain.